File: backend/initDB.js
import fs from 'fs';
import db from './db.js';
import path from 'path';
import { fileURLToPath } from 'url';

const __filename = fileURLToPath(import.meta.url);
const __dirname = path.dirname(__filename);

async function initDB() {
  try {
    const sql = fs.readFileSync(path.join(__dirname, 'init.sql'), 'utf-8');

    // Run the entire SQL file as a single statement (PostgreSQL handles this fine)
    await db.query(sql);

    console.log('Database initialized successfully from init.sql!');
    process.exit(0);
  } catch (error) {
    console.error('Error initializing database:', error);
    process.exit(1);
  }
}

initDB();

File: backend_flask/init_db.py
"""Initialize the database by running init.sql against MySQL."""

import mysql.connector
import config
import os


def init_db():
    conn = mysql.connector.connect(
        host=config.DB_HOST,
        user=config.DB_USER,
        password=config.DB_PASSWORD,
    )
    cursor = conn.cursor()

    sql_path = os.path.join(os.path.dirname(__file__), "init.sql")
    with open(sql_path, "r", encoding="utf-8") as f:
        sql = f.read()

    for statement in sql.split(";"):
        statement = statement.strip()
        if statement:
            try:
                cursor.execute(statement)
            except mysql.connector.Error as e:
                print(f"Warning: {e}")

    conn.commit()
    cursor.close()
    conn.close()
    print("Database initialized successfully from init.sql!")


if __name__ == "__main__":
    init_db()
